zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/incident/IncidentResolveProcessor.java

```diff
@@ -72,10 +74,17 @@ public void processRecord(final TypedRecord<IncidentRecord> command) {
       return;
     }
 
+    final long jobKey = incident.getJobKey();
+    if (isJobRelatedIncident(jobKey) && jobState.getJob(jobKey).getRetries() <= 0) {
```

Comment:
❓ could `jobState.getJob(jobKey)` ever return `null`? 

💭 I know this would have thrown also at `jobActivationBehavior.publishWork(jobKey, failedJobRecord)`, but just raising the question to think about null safety here

Reply:
I am adding a null check, could you please take a look?

Reply:
@mustafadagher @ana-vinogradova-camunda If retrieving job from the state returns null, that points to a bug in our system. Because, the job key is set to the incident record is expected to be found in the state. Other processors or any other code in our system should not remove this job from the state. Therefore, we shouldn't cover the case where the job is not found with the usual rejection.

Reply:
@berkaycanbc should the check be present in `IncindentResolverProcessor` then? If not, then it is another issue, I'd say, unrelated to what is fixed here.

Reply:
@ana-vinogradova-camunda it is not an issue existing. If there would appear to be an issue where the job is not found, an exception should be thrown and the error message should be observed on our logs. Then, we can create an issue for that. For now, removing the null check should be sufficient.

Reply:
@berkaycanbc thank you for the clarification!